code/modules/xenoarcheaology/effects/cellcharge.dm

```diff
@@ -6,13 +6,33 @@
 
 
 /datum/artifact_effect/uncommon/cellcharge/DoEffectTouch(mob/living/user)
-	if (user)
-		if (istype(user, /mob/living/silicon/robot))
-			var/mob/living/silicon/robot/R = user
-			for (var/obj/item/cell/D in R.contents)
-				D.charge += rand() * 100 + 50
-				to_chat(R, "<font color='blue'>SYSTEM ALERT: Large energy boost detected!</font>")
+	if(!user)
+		return
+	var/atom/holder = get_master_holder()
+	if (!holder)
+		return
+	var/turf/turf = get_turf(user)
+	if (!turf)
+		return
+
+	var/messaged_robots
+	for(var/mob/living/L in range(effectrange, turf))
+		var/obj/item/cell/C = L.get_cell()
```

Comment:
`get_cell()` is a proc on `/atom/movable` - can we iterate over those and do our own typechecking for the messaging stuff?